resip/stack/Uri.cxx

```diff
@@ -748,7 +703,95 @@ Uri::operator<(const Uri& other) const
       return false;
    }
 
-   return mPort < other.mPort;
+   if (mPort < other.mPort)
+   {
+      return true;
+   }
+
+   if (mPort > other.mPort)
+   {
+      return false;
+   }
+
+   for (ParameterList::const_iterator it = mParameters.begin(); it != mParameters.end(); ++it)
+      {
+         Parameter* otherParam = other.getParameterByEnum((*it)->getType());
+
+         if (otherParam)
+         {
+            if (Uri::compareUriParametersLessThan(*it, otherParam)) {
+               return true;
+            }
+
+            if (Uri::compareUriParametersLessThan(otherParam, *it)) {
+               return false;
+            }
+         }
+         else if (Uri::isSignificantUriParameter((*it)->getType()))
+         {
+            return true;  // Significant URI params should not be ignored.
+         }
+      }
+
+   OrderUnknownParameters orderUnknown;
+
+#if defined(__SUNPRO_CC) || defined(WIN32) || defined(__sun__)
+   using std::set<Parameter*, OrderUnknownParameters> ParameterSet;
+   ParameterSet unA, unB;
+
+   for (ParameterList::const_iterator i = mUnknownParameters.begin();
+        i != mUnknownParameters.end(); ++i)
+   {
+      unA.insert(*i);
+   }
+   for (ParameterList::const_iterator i = other.mUnknownParameters.begin();
+        i != other.mUnknownParameters.end(); ++i)
+   {
+      unB.insert(*i);
+   }
+#else
+   ParameterList unA = mUnknownParameters;
+   ParameterList unB = other.mUnknownParameters;
+
+   sort(unA.begin(), unA.end(), orderUnknown);
+   sort(unB.begin(), unB.end(), orderUnknown);
+#endif
+
```

Comment:
Can you please add some comments in the code explaining what this section of code is doing?  ie:  Why build up 2 new sets?  Why is code different for different platforms?  And a general explanation of the logic below.  :)  Thank you in advance.

Reply:
Added comments and renamed some variables to make it easier to understand.
This part of the code mirrors the one from `Uri::operator==`.